app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Animal) -> None:
+        if not isinstance(herbivore, Carnivore) and not herbivore.hidden:
```

Comment:
The `bite` method should ensure that the `herbivore` parameter is an instance of `Herbivore`, not just any `Animal`. Consider changing the condition to `isinstance(herbivore, Herbivore)` to ensure that only `Herbivore` instances can be bitten.